src/push/kibana_api.ts

```diff
@@ -135,6 +135,13 @@ export async function getVersion(options: PushOptions) {
     method: 'GET',
     auth: options.auth,
   });
+
+  if (!data.version || typeof data.version?.number === 'undefined') {
+    throw Error(
+      'Unable to retrieve version. Are your auth credentials correct?'
```

Comment:
not sure how helpful this message would be to the user. Maybe something like `API credentials is incorrect, unable to push monitors`?.

Reply:
Appreciate the feedback here, the copy was meant to be placeholder so the suggestions are welcome. That said, I just pushed [7cc61ff](https://github.com/elastic/synthetics/pull/821/commits/7cc61ff303bf825940f2ce39ca14c00c466a1ccd), and it makes a custom error handler no longer needed. We will receive a `401` error _and_ get the version info we want, which is a far cleaner and more correct solution. I'll update the description to reflect it.